cmd/kube-proxy/app/server.go

```diff
@@ -366,16 +367,19 @@ func badCIDRs(cidrs []string, wrongFamily netutils.IPFamily) bool {
 	return true
 }
 
-// badBindAddress returns true if bindAddress is an "IP:port" string where IP is a
-// non-zero IP of wrongFamily.
-func badBindAddress(bindAddress string, wrongFamily netutils.IPFamily) bool {
-	if host, _, _ := net.SplitHostPort(bindAddress); host != "" {
-		ip := netutils.ParseIPSloppy(host)
-		if ip != nil && netutils.IPFamilyOf(ip) == wrongFamily && !ip.IsUnspecified() {
-			return true
+// badBindAddresses returns true if cidrs is a non-empty list of CIDRs, all of wrongFamily.
+// Unspecified addresses '0.0.0.0' and '::' are not treated as part of either family.
+func badBindAddresses(addresses []string, wrongFamily netutils.IPFamily) bool {
+	if len(addresses) == 0 {
+		return false
+	}
+	for _, address := range addresses {
+		ip, _, _ := netutils.ParseCIDRSloppy(address)
+		if netutils.IPFamilyOf(ip) != wrongFamily || ip.IsUnspecified() {
```

Comment:
you need to check that the mask length is 0 too; `0.0.0.0/32` should not be considered a valid IPv6 CIDR